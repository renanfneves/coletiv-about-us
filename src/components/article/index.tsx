import React, { FC } from 'react';

import { Container, Image, Subject, SubTitle, Content } from './styles';

interface Props {
  image: string;
  alt: string;
  subject: string;
  subTitle: string;
  reverse?: boolean;
}

const Article: FC<Props> = ({ image, alt, subject, subTitle, children, reverse }) => (
  <Container reverse={reverse}>
    <Image src={image} alt={alt} />
    <Content>
      <Subject>{subject}</Subject>
      <SubTitle>{subTitle}</SubTitle>
      {children}
    </Content>
  </Container>
);

export default Article;