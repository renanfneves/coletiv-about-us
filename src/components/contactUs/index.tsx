import React, { FC } from 'react';

import { Container, Title, SubTitle, Text, ContactButton } from './styles';

const contactUs: FC = () => {
  return (
    <Container>
      <Title>CONTACT US</Title>
      <SubTitle>From humble beginnings</SubTitle>
      <Text>We know the challenges of creating a business. Want us to help create yours?</Text>
      <ContactButton href="https://coletiv.com/contact">Get In Touch</ContactButton>
    </Container>
  );
}

export default contactUs;