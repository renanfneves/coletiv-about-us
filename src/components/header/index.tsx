import React, { FC, useCallback, useState } from 'react';

import {
  Container,
  LogoButton,
  WebNav,
  MobileNav,
  WebLink,
  TalkToUs,
  BurgerIcon,
  StyledModal,
  ModalHeader,
  CloseButton,
  ModalLinks,
  SubMenu,
  ModalLink,
  DropdownSubmenu,
} from './styles';

import { Logo } from '../../assets';
import Dropdown from '../dropdown';

const Header: FC = () => {
  const [modalIsOpen, setIsOpen] = useState(false);

  const closeModal = useCallback(() => {
    setIsOpen(false);
  }, []);

  const openModal = useCallback(() => {
    setIsOpen(true);
  }, []);


  return (
    <Container>
      <StyledModal isOpen={modalIsOpen} ariaHideApp={false}>
        <ModalHeader>
          <LogoButton onClick={() => { window.location.href = "https://coletiv.com/"}}>
            <Logo />
          </LogoButton>
          <CloseButton onClick={closeModal}>X</CloseButton>
        </ModalHeader>
        <ModalLinks>
          <ModalLink href="https://coletiv.com/showcase/">Showcase</ModalLink>
          <ModalLink href="https://coletiv.com/showcase/">Services</ModalLink>
          <SubMenu href="https://coletiv.com/services/ux-ui-design/">UX/UI Design</SubMenu>
          <SubMenu href="https://coletiv.com/services/web-development/">Web Development</SubMenu>
          <SubMenu href="https://coletiv.com/services/android-app-development/">Android Development</SubMenu>
          <SubMenu href="https://coletiv.com/services/ios-app-development/">iOS Development</SubMenu>
          <SubMenu href="https://coletiv.com/services/flutter-development">Flutter Development</SubMenu>
          <SubMenu href="https://coletiv.com/services/elixir-backend/">Backend Development</SubMenu>
          <ModalLink href="https://coletiv.com/blog/">Blog</ModalLink>
          <ModalLink href="/" style={{ fontWeight: 700, color: 'black' }}>About Us</ModalLink>
        </ ModalLinks>
      </StyledModal>
      <LogoButton onClick={() => { window.location.href = "https://coletiv.com/"}}>
        <Logo />
      </LogoButton>
      <WebNav>
        <WebLink href="https://coletiv.com/showcase/">Showcase</WebLink>
        <Dropdown 
          label="Services"
          onClick={() => { window.location.href = "https://coletiv.com/services/"}}
        >
          <nav>
            <DropdownSubmenu href="https://coletiv.com/services/ux-ui-design/">UX/UI Design</DropdownSubmenu>
            <DropdownSubmenu href="https://coletiv.com/services/web-development/">Web Development</DropdownSubmenu>
            <DropdownSubmenu href="https://coletiv.com/services/android-app-development/">Android Development</DropdownSubmenu>
            <DropdownSubmenu href="https://coletiv.com/services/ios-app-development/">iOS Development</DropdownSubmenu>
            <DropdownSubmenu href="https://coletiv.com/services/flutter-development">Flutter Development</DropdownSubmenu>
            <DropdownSubmenu href="https://coletiv.com/services/elixir-backend/">Backend Development</DropdownSubmenu>
          </nav>
        </Dropdown>
        <WebLink href="https://coletiv.com/blog/">Blog</WebLink>
        <WebLink href="/" style={{ fontWeight: 700, color: 'black' }}>About Us</WebLink>
        <TalkToUs href="https://coletiv.com/contact">Talk To Us</TalkToUs>
      </WebNav>
      <MobileNav>
        <TalkToUs href="https://coletiv.com/contact">Talk To Us</TalkToUs>
        <BurgerIcon onClick={openModal}>
          <span />
          <span />
          <span />
        </BurgerIcon>
      </MobileNav>
    </Container>
  );
};

export default Header;