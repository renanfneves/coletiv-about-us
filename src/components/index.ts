export { default as Header } from './header';
export { default as Article } from './article';
export { default as Paragraph } from './paragraph';
export { default as ContactUs } from './contactUs';
export { default as Footer } from './footer';
export { default as Dropdown } from './dropdown';