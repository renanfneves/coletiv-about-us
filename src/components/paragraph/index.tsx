import React, { FC } from 'react';

import { Container } from './styles';

interface Props {
  italic?: boolean;
}

const Paragraph: FC<Props> = ({ italic, children }) => {
  return <Container italic={italic}>{children}</Container>;
}

export default Paragraph;