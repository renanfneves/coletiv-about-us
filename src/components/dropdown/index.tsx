import React, { FC, useCallback, useState } from 'react';
import { DownArrow } from 'src/assets';

import { Container, Label, Content } from './styles';

interface Props {
  label: string;
  onClick: () => void;
}

const Dropdown: FC<Props> = ({ label, children, onClick }) => {
  const [isOpen, setIsOpen] = useState(false);

  const toggleDropdown = useCallback((open: boolean) => {
    setIsOpen(open);
  }, []);

  return (
    <Container
      onMouseOver={() => toggleDropdown(true)}
      onMouseLeave={() => toggleDropdown(false)}
      onClick={onClick}
    >
      <Label>{label}</Label>
      <DownArrow />
      {isOpen && <Content>{children}</Content>}
    </Container>
  );
}

export default Dropdown;