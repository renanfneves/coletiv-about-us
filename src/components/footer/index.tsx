import React, { FC } from 'react';
import { CompleteLogo, Dribbble, Facebook, Instagram, LinkedIn, Twitter } from 'src/assets';

import { Container, Background, Main, PrivacyWrapper, FooterNav, Socials, Copyright, ContactsBanner, Slogan, ContactsWrapper, ContactsContent } from './styles';

const footer: FC = () => {
  return (
    <Container>
      <ContactsBanner>
        <Slogan>
          <h3>Your full-service digital agency</h3>
        </Slogan>
        <ContactsWrapper>
          <ContactsContent>
            <h3>Contacts</h3>
            <a target="_self" href="mailto:humans@coletiv.com" rel="noopener noreferrer">humans@coletiv.com</a>
            <a target="_self" href="tel:+351914541645" rel="noopener noreferrer">+351 914 541 645</a>
          </ContactsContent>
        </ContactsWrapper>
      </ContactsBanner>
      <Background>
        <Main>
          <div>
            <a href="https://coletiv.com/">
              <CompleteLogo />
            </a>
            <PrivacyWrapper>
              <span>@ 2020 Coletiv</span>
              <a href="https://coletiv.com/legal">{"Privacy & Policy"}</a>
            </PrivacyWrapper>
          </div>
          <FooterNav>
            <h3>Company</h3>
            <ul>
              <li><a href="https://coletiv.com/showcase/">Showcase</a></li>
              <li><a href="https://coletiv.com/services/">Services</a></li>
              <li><a href="https://coletiv.com/blog/">Blog</a></li>
              <li><a href="/">About Us</a></li>
            </ul>
          </FooterNav>
          <FooterNav>
          <h3>Contacts</h3>
          <ul>
            <li>
              <a 
                target="_self"
                href="mailto:humans@coletiv.com"
                rel="noopener noreferrer">
                  Email humans@coletiv.com
              </a>
            </li>
            <li>
              <a target="_self"
              href="tel:+351914541645"
              rel="noopener noreferrer"
              >
                Phone +351 914 541 645
              </a>
            </li>
          </ul>
          </FooterNav>
          <Socials>
          <h3>Social</h3>
          <ul>
            <li>
              <a
                href="//linkedin.com/company/coletiv"
                title="See Coletiv on LinkedIn"
                target="_blank" rel="noopener noreferrer"
              >
                <LinkedIn />
              </a>
            </li>

            <li>
              <a
                href="//twitter.com/coletivstudio"
                title="See Coletiv on Twitter"
                target="_blank" rel="noopener noreferrer"
              >
                <Twitter />
              </a>
            </li>

            <li>
              <a
                href="//facebook.com/wearecoletiv"
                title="See Coletiv on Facebook"
                target="_blank"
                rel="noopener noreferrer"
              >
                <Facebook />
              </a>
            </li>

            <li>
              <a
                href="//instagram.com/coletiv"
                title="See Coletiv on Instagram"
                target="_blank"
                rel="noopener noreferrer"
              >
                <Instagram />
              </a>
            </li>

            <li>
              <a
                href="//dribbble.com/Coletiv"
                title="See Coletiv on Dribbble"
                target="_blank"
                rel="noopener noreferrer"
              >
                <Dribbble />
              </a>
            </li>
          </ul>
        </Socials>
        </Main>
        <Copyright>
          <a href="https://coletiv.com//legal">
            Copyright Â© Coletiv Studio, Lda. 2020 All rights reserved. 
            If you are bored and want to read some legal bits, see our legal page.
          </a>
        </Copyright>
      </Background>
    </Container>
  );
}

export default footer;