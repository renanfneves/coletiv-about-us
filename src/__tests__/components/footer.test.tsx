import React from 'react';
import { shallow } from 'enzyme';
import { Footer } from '../../components';

test('It renders the footer component', () => {
  const component = shallow(<Footer />);
  expect(component).toMatchSnapshot();
});