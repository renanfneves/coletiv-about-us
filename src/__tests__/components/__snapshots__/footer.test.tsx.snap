// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the footer component 1`] = `
<styled.footer>
  <styled.article>
    <styled.article>
      <h3>
        Your full-service digital agency
      </h3>
    </styled.article>
    <styled.article>
      <styled.article>
        <h3>
          Contacts
        </h3>
        <a
          href="mailto:humans@coletiv.com"
          rel="noopener noreferrer"
          target="_self"
        >
          humans@coletiv.com
        </a>
        <a
          href="tel:+351914541645"
          rel="noopener noreferrer"
          target="_self"
        >
          +351 914 541 645
        </a>
      </styled.article>
    </styled.article>
  </styled.article>
  <styled.footer>
    <styled.div>
      <div>
        <a
          href="https://coletiv.com/"
        >
          <CompleteLogo />
        </a>
        <styled.p>
          <span>
            @ 2020 Coletiv
          </span>
          <a
            href="https://coletiv.com/legal"
          >
            Privacy & Policy
          </a>
        </styled.p>
      </div>
      <styled.nav>
        <h3>
          Company
        </h3>
        <ul>
          <li>
            <a
              href="https://coletiv.com/showcase/"
            >
              Showcase
            </a>
          </li>
          <li>
            <a
              href="https://coletiv.com/services/"
            >
              Services
            </a>
          </li>
          <li>
            <a
              href="https://coletiv.com/blog/"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              About Us
            </a>
          </li>
        </ul>
      </styled.nav>
      <styled.nav>
        <h3>
          Contacts
        </h3>
        <ul>
          <li>
            <a
              href="mailto:humans@coletiv.com"
              rel="noopener noreferrer"
              target="_self"
            >
              Email humans@coletiv.com
            </a>
          </li>
          <li>
            <a
              href="tel:+351914541645"
              rel="noopener noreferrer"
              target="_self"
            >
              Phone +351 914 541 645
            </a>
          </li>
        </ul>
      </styled.nav>
      <styled.nav>
        <h3>
          Social
        </h3>
        <ul>
          <li>
            <a
              href="//linkedin.com/company/coletiv"
              rel="noopener noreferrer"
              target="_blank"
              title="See Coletiv on LinkedIn"
            >
              <LinkedIn />
            </a>
          </li>
          <li>
            <a
              href="//twitter.com/coletivstudio"
              rel="noopener noreferrer"
              target="_blank"
              title="See Coletiv on Twitter"
            >
              <LinkedIn />
            </a>
          </li>
          <li>
            <a
              href="//facebook.com/wearecoletiv"
              rel="noopener noreferrer"
              target="_blank"
              title="See Coletiv on Facebook"
            >
              <LinkedIn />
            </a>
          </li>
          <li>
            <a
              href="//instagram.com/coletiv"
              rel="noopener noreferrer"
              target="_blank"
              title="See Coletiv on Instagram"
            >
              <LinkedIn />
            </a>
          </li>
          <li>
            <a
              href="//dribbble.com/Coletiv"
              rel="noopener noreferrer"
              target="_blank"
              title="See Coletiv on Dribbble"
            >
              <LinkedIn />
            </a>
          </li>
        </ul>
      </styled.nav>
    </styled.div>
    <styled.footer>
      <a
        href="https://coletiv.com//legal"
      >
        Copyright Â© Coletiv Studio, Lda. 2020 All rights reserved. If you are bored and want to read some legal bits, see our legal page.
      </a>
    </styled.footer>
  </styled.footer>
</styled.footer>
`;
